packages/checkout/widgets-lib/src/components/Transactions/Transactions.tsx

```diff
@@ -103,19 +106,40 @@ export function Transactions({ checkout }: TransactionsProps) {
       console.error(e);
       return [];
     }
-  }, [checkout, provider, walletAddress]);
+  }, [checkout, from]);
 
   const updateAndConnectProvider = useCallback(async (walletProviderName: WalletProviderName) => {
+    track({
+      userJourney: UserJourney.BRIDGE,
+      screen: 'EmptyStateNotConnected',
+      control: 'WalletProvider',
+      controlType: 'Select',
+      extras: {
+        walletProviderName,
+      },
+    });
     try {
-      const localProvider = await createAndConnectToProvider(checkout, walletProviderName);
+      const localProvider = await createAndConnectToProvider(checkout, walletProviderName, true);
       const address = await localProvider?.getSigner().getAddress() ?? '';
-      setProvider(localProvider);
-      setWalletAddress(address.toLowerCase());
+      setTxs([]);
+      bridgeDispatch({
+        payload: {
+          type: BridgeActions.SET_WALLETS_AND_NETWORKS,
+          from: {
+            web3Provider: localProvider,
+            walletAddress: address.toLowerCase(),
+            network: ChainId.SEPOLIA, // TODO: figure out what to do with network here
```

Comment:
is this TODO for a separate PR?

Reply:
Thanks Mikhala, good catch. Have fixed this now to just get the provider's network. This actually isn't used in the claim withdrawal anyway but good to have it as the correct value.